src/main/java/com/habicus/core/service/Goal/GoalService.java

```diff
@@ -37,4 +38,15 @@
     List<Goal> goals = goalRepository.findGoalsByUserId(id);
     return goals;
   }
+
+  /**
+   * Allows a {@link com.habicus.core.model.User} to persist a {@link Goal} object into the DB
+   *
+   * @param goal
+   * @return
+   */
+  public Goal saveUserGoal(Goal goal) {
+    if (goal != null) return goalRepository.save(goal);
```

Comment:
Eventually want to add more validation to the goal, ensuring all items are filled out (e.g. units shouldn't be null)

Reply:
Agreed. I think this can exist at the DAO level and also have object validation in the service layer.